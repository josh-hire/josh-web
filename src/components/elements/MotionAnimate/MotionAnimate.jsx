import { useEffect } from 'react';
import PropTypes from 'prop-types';
import clsx from 'clsx';
import { useInView } from 'react-intersection-observer';
import 'animate.css/source/zooming_entrances/zoomIn.css';
import 'animate.css/source/zooming_entrances/zoomInDown.css';
import 'animate.css/source/zooming_entrances/zoomInUp.css';
import 'animate.css/source/zooming_exits/zoomOut.css';
import 'animate.css/source/zooming_exits/zoomOutDown.css';
import 'animate.css/source/zooming_exits/zoomOutUp.css';
import 'animate.css/source/fading_entrances/fadeIn.css';
import 'animate.css/source/fading_entrances/fadeInDown.css';
import 'animate.css/source/fading_entrances/fadeInLeft.css';
import 'animate.css/source/fading_entrances/fadeInRight.css';
import 'animate.css/source/fading_entrances/fadeInUp.css';
import 'animate.css/source/fading_entrances/fadeInTopLeft.css';
import 'animate.css/source/fading_entrances/fadeInTopRight.css';
import 'animate.css/source/fading_entrances/fadeInBottomLeft.css';
import 'animate.css/source/fading_entrances/fadeInBottomRight.css';
import 'animate.css/source/attention_seekers/bounce.css';
import 'animate.css/source/attention_seekers/pulse.css';
import 'animate.css/source/bouncing_entrances/bounceIn.css';
import 'animate.css/source/bouncing_entrances/bounceInDown.css';
import 'animate.css/source/bouncing_entrances/bounceInLeft.css';
import 'animate.css/source/bouncing_entrances/bounceInRight.css';
import 'animate.css/source/flippers/flipInX.css';
import 'animate.css/source/flippers/flipInY.css';
import 'animate.css/source/flippers/flipOutY.css';
import 'animate.css/source/rotating_entrances/rotateIn.css';
import 'animate.css/source/rotating_exits/rotateOut.css';
import 'animate.css/source/specials/jackInTheBox.css';

import styles from './styles.module.scss';

const MotionAnimate = ({
  children,
  className,
  animate,
  key,
  autoAnimate,
  trigger,
  style,
  onActiveChange,
}) => {
  const { ref: HeroRef, inView } = useInView();
  useEffect(() => {
    if (onActiveChange) {
      onActiveChange(inView);
    }
  }, [inView]);
  return (
    <div
      className={clsx(
        styles.root,
        className,
        autoAnimate && inView ? animate : '',
        trigger ? animate : '',
      )}
      key={key || 'PT_tester.id'}
      ref={HeroRef}
      style={{ ...style }}>
      {children}
    </div>
  );
};

MotionAnimate.propTypes = {
  animate: PropTypes.string,
  autoAnimate: PropTypes.bool,
  children: PropTypes.node,
  className: PropTypes.string,
  key: PropTypes.string,
  onActiveChange: PropTypes.func,
  style: PropTypes.object,
  trigger: PropTypes.bool,
};

MotionAnimate.defaultProps = {
  animate: '',
  autoAnimate: false,
  children: null,
  className: '',
  key: 'PT_tester.id',
  onActiveChange: null,
  style: {},
  trigger: false,
};

export { MotionAnimate };
