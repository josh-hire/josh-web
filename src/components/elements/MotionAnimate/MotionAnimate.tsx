import { useEffect, ReactNode, CSSProperties } from 'react';
import clsx from 'clsx';
import { useInView } from 'react-intersection-observer';
import 'animate.css/source/zooming_entrances/zoomIn.css';
import 'animate.css/source/zooming_entrances/zoomInDown.css';
import 'animate.css/source/zooming_entrances/zoomInUp.css';
import 'animate.css/source/zooming_exits/zoomOut.css';
import 'animate.css/source/zooming_exits/zoomOutDown.css';
import 'animate.css/source/zooming_exits/zoomOutUp.css';
import 'animate.css/source/fading_entrances/fadeIn.css';
import 'animate.css/source/fading_entrances/fadeInDown.css';
import 'animate.css/source/fading_entrances/fadeInLeft.css';
import 'animate.css/source/fading_entrances/fadeInRight.css';
import 'animate.css/source/fading_entrances/fadeInUp.css';
import 'animate.css/source/fading_entrances/fadeInTopLeft.css';
import 'animate.css/source/fading_entrances/fadeInTopRight.css';
import 'animate.css/source/fading_entrances/fadeInBottomLeft.css';
import 'animate.css/source/fading_entrances/fadeInBottomRight.css';
import 'animate.css/source/attention_seekers/bounce.css';
import 'animate.css/source/attention_seekers/pulse.css';
import 'animate.css/source/bouncing_entrances/bounceIn.css';
import 'animate.css/source/bouncing_entrances/bounceInDown.css';
import 'animate.css/source/bouncing_entrances/bounceInLeft.css';
import 'animate.css/source/bouncing_entrances/bounceInRight.css';
import 'animate.css/source/flippers/flipInX.css';
import 'animate.css/source/flippers/flipInY.css';
import 'animate.css/source/flippers/flipOutY.css';
import 'animate.css/source/rotating_entrances/rotateIn.css';
import 'animate.css/source/rotating_exits/rotateOut.css';
import 'animate.css/source/specials/jackInTheBox.css';
import styles from './styles.module.scss';

interface MotionAnimateProps {
  children?: ReactNode;
  className?: string;
  animate?: string;
  key?: string;
  autoAnimate?: boolean;
  trigger?: boolean;
  style?: CSSProperties;
  onActiveChange?: (inView: boolean) => void;
}

const MotionAnimate: React.FC<MotionAnimateProps> = ({
  children,
  className,
  animate = '',
  autoAnimate = false,
  trigger = false,
  style = {},
  onActiveChange,
}) => {
  const { ref: HeroRef, inView } = useInView();

  useEffect(() => {
    if (onActiveChange) {
      onActiveChange(inView);
    }
  }, [inView, onActiveChange]);

  return (
    <div
      className={clsx(
        styles.root,
        className,
        autoAnimate && inView ? animate : '',
        trigger ? animate : '',
      )}
      ref={HeroRef}
      style={style}
    >
      {children}
    </div>
  );
};

export { MotionAnimate };
