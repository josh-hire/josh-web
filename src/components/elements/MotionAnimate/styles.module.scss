.root{
  animation-duration: 1.2s;
  padding: 0;
  margin: 0;
}
