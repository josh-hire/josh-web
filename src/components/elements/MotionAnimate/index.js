import { MotionAnimate } from './MotionAnimate';

export default MotionAnimate;
