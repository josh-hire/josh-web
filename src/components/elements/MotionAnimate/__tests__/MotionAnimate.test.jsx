/* eslint-disable react/jsx-handler-names */
import '@testing-library/jest-dom';
import { MotionAnimate as Component } from '../MotionAnimate';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import * as reactIntersectionMock from 'react-intersection-observer';

beforeEach(() => {
  const mockIntersectionObserver = jest.fn();
  mockIntersectionObserver.mockReturnValue({
    observe: jest.fn,
    unobserve: jest.fn,
    disconnect: jest.fn
  });
  window.IntersectionObserver = mockIntersectionObserver;
});

describe('src/components/elements/MotionAnimate', ()=>{
  test('MotionAnimate rtl',()=>{
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
    
  });
  test('MotionAnimate rtl onActiveChange',()=>{

    reactIntersectionMock.InView = jest.fn().mockReturnValue({ inView: true });

    const { asFragment } = render(<Component autoAnimate onActiveChange={jest.fn} trigger />);
    expect(asFragment()).toMatchSnapshot();
    
  });
});
