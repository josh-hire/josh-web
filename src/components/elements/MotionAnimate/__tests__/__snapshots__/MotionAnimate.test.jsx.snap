// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/MotionAnimate MotionAnimate rtl 1`] = `
<DocumentFragment>
  <div
    class="root"
  />
</DocumentFragment>
`;

exports[`src/components/elements/MotionAnimate MotionAnimate rtl onActiveChange 1`] = `
<DocumentFragment>
  <div
    class="root"
  />
</DocumentFragment>
`;
